src/cli/commands/import.js

```diff
@@ -254,7 +254,10 @@ class ImportPackageResolver extends PackageResolver {
     }
   }
 
-  async init(deps: DependencyRequestPatterns, isFlat: boolean): Promise<void> {
+  async init(
+    deps: DependencyRequestPatterns,
+    {isFlat, isFrozen}: {isFlat: boolean, isFrozen?: boolean},
```

Comment:
Shouldn't `isFlat` be also a `isFlat?` in its type? I'm not really familiar with Flow, though. On a different note, you can add a default value (` = {}` at the end of the parameter declaration) so that you don't have to explicitely pass an empty object (like [here](https://github.com/yarnpkg/yarn/pull/3604/files#diff-d8db564fe8c688cb48a7eea6336f3895L147)). That might require to move the `workspaceLayout` parameter before the options.